### PR TITLE
LG-2502, LG-2488: Refresh code

### PR DESCRIPTION
**Pull request by shaneeza**
_Note_: I'm not 100% sure the colors are correct. I spoke with Dave and he said for the code syntax I only needed to change the white text. So for example, `datetime` in the storybook example. I plan on confirming with him when he's back. 
Screen Shot 2022-10-28 at 3 16 39 PM

## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-2502](https://jira.mongodb.org/browse/LG-2502)
🎟 _Jira ticket:_ [LG-2488](https://jira.mongodb.org/browse/LG-2488)
Screen Shot 2022-10-28 at 3 17 06 PM

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/code/src/CopyButton.tsx [hunk 1]
```diff
@@ -2,69 +2,64 @@ import React, { useState, useEffect } from 'react';
...
-    color: ${palette.white};
+    &:focus,
+    &:hover {
+      color: ${palette.white};
 
-    &:before {
-      background-color: ${palette.green.dark1};
+      &:before {
+        background-color: ${palette.green.dark1};
+      }
```
Comment (TheSonOfThomp) [inline packages/code/src/CopyButton.tsx:24]: From Figma, I think this is supposed to be `green.base` 
Screen Shot 2022-11-02 at 4 41 36 PM
Screen Shot 2022-11-02 at 4 41 32 PM

File: packages/code/src/CodeContext.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export function useCodeContext() {
+  const { theme, darkMode } = useContext(CodeContext);
+
+  return { theme, darkMode };
+}
+
+export const CodeContext = createContext<CodeData>({
+  theme: Theme.Light,
+  darkMode: false,
+});
```
Comment (TheSonOfThomp) [inline packages/code/src/CodeContext.tsx:18]: Could we use a new `LeafyGreenProvider` instead, or do you anticipate needing other properties in the future?

File: packages/code/src/Syntax.tsx [hunk 3]
```diff
@@ -98,14 +96,14 @@ function Syntax({
...
   const baseFontSize = useBaseFontSize();
-  // TODO: Refresh - remove darkMode logic
-  const codeFontSize = baseFontSize === 14 ? 13 : darkMode ? 16 : 15;
-
+  // TODO: remove 14 check when useBaseFontSize is updated
+  const codeSize = baseFontSize === 14 ? '1' : '2';
   const codeFontStyles = css`
-    font-size: ${codeFontSize}px;
+    font-size: ${typeScales[`code${codeSize}`].fontSize}px;
+    line-height: ${typeScales[`code${codeSize}`].lineHeight}px;
```
Comment (TheSonOfThomp) [inline packages/code/src/Syntax.tsx:106]: This is a little confusing. I think using the ternary to select the typescale would be a little clearer.
```js
const typeScale = (baseFontSize === 14) ? typeScales['code1'] : typeScales['code2']
```

Comment (TheSonOfThomp) [reply packages/code/src/Syntax.tsx:106]: (we should have a utility that converts `baseFontSize` to a relevant `typeScale`

File: packages/code/src/renderingPlugin.tsx [hunk 4]
```diff
@@ -114,14 +114,15 @@ function getHighlightedRowStyle(darkMode: boolean) {
...
+      100,
+      '#21313C',
+    )})`;
+    borderColor = '#21313C';
   } else {
-    backgroundColor = uiColors.yellow.light3;
+    backgroundColor = '#FEF7E3';
     backgroundImage = 'none';
-    borderColor = uiColors.yellow.light2;
+    borderColor = '#FEF2C8';
```
Comment (TheSonOfThomp) [inline packages/code/src/renderingPlugin.tsx:125]: If we use a hex value that's off palette, we should comment why

File: packages/code/src/renderingPlugin.tsx [hunk 5]
```diff
@@ -167,10 +168,8 @@ export function LineTableRow({
   darkMode,
   children,
 }: LineTableRowProps) {
-  const numberColor = darkMode ? uiColors.gray.dark1 : palette.gray.dark1;
-  const highlightedNumberColor = darkMode
-    ? uiColors.gray.light2
-    : palette.yellow.dark2;
+  const numberColor = darkMode ? '#5D6C74' : palette.gray.dark1;
+  const highlightedNumberColor = darkMode ? '#E7EEEC' : palette.yellow.dark2;
```
Comment (TheSonOfThomp) [inline packages/code/src/renderingPlugin.tsx:172]: Same here

File: packages/code/src/types.ts [hunk 6]
```diff
@@ -118,16 +111,23 @@ export type CodeProps = Omit<
   onCopy?: Function;
 
   /**
-   * Custom action buttons.
+   * Custom action buttons. Should be an array of `IconButton`.
    *
    */
-  customActionButtons?: Array<React.ReactNode>;
+  customActionButtons?: Array<React.ReactElement>;
```
Comment (TheSonOfThomp) [inline packages/code/src/types.ts:117]: Why `ReactElement` over `ReactNode`?

Comment (TheSonOfThomp) [reply packages/code/src/types.ts:117]: Also we should add a custom TSDoc annotation here:
`@type <IconButton />[]`

File: yarn.lock [hunk 7]
```diff
@@ -2293,6 +2293,18 @@
     "@leafygreen-ui/ripple" "^1.1.1"
     "@leafygreen-ui/tokens" "^0.5.3"
 
+"@leafygreen-ui/button@^17.0.0":
```
Comment (TheSonOfThomp) [inline yarn.lock:2296]: Why is `yarn.lock` changing? There are no changes to a `package.json`

**Review by TheSonOfThomp — CHANGES_REQUESTED** [review]
Just a few small nits

File: packages/code/src/CodeContext.tsx (hunk 2 above)
Comment (shaneeza) [reply packages/code/src/CodeContext.tsx:18]: I don't anticipate anything else. I'll try it out!

File: packages/code/src/types.ts (hunk 6 above)
Comment (shaneeza) [reply packages/code/src/types.ts:117]: I was thinking that `ReactElement` was a little stricter. We already have a check to make sure that the customActionButtons are `IconButton` so I didn't see the harm in changing it.

File: packages/code/src/Code.tsx [hunk 8]
```diff
@@ -35,7 +34,7 @@ const lineHeight = 24;
 
 const wrapperStyle: Record<Theme, string> = {
   [Theme.Light]: css`
-    border: 1px solid ${variantColors[Theme.Light][1]};
+    border: 1px solid ${variantColors.light[1]};
```
Comment (TheSonOfThomp) [inline packages/code/src/Code.tsx:37]: I think using `[Theme.Light]` is technically correct—we should prefer to use use the enum where we can

File: packages/code/src/types.ts (hunk 6 above)
Comment (TheSonOfThomp) [reply packages/code/src/types.ts:118]: Oh, yeah, that makes sense 👍

File: packages/code/src/renderingPlugin.tsx [hunk 9]
```diff
@@ -112,16 +111,12 @@ function getHighlightedRowStyle(darkMode: boolean) {
...
-    }, ${transparentize(100, uiColors.gray.dark3)})`;
-    borderColor = uiColors.gray.dark3;
+    backgroundImage = `linear-gradient(90deg, ${palette.gray.dark3}, ${palette.black})`;
+    borderColor = palette.gray.dark3;
   } else {
-    backgroundColor = uiColors.yellow.light3;
+    backgroundColor = '#FEF7E3'; // not updating to palette yet, currently sticking with uiColor for accessibility reasons
     backgroundImage = 'none';
-    borderColor = uiColors.yellow.light2;
+    borderColor = '#FEF2C8'; // not updating to palette yet, currently sticking with uiColor for accessibility reasons
```
Comment (TheSonOfThomp) [inline packages/code/src/renderingPlugin.tsx:119]: In that case, maybe label with `// TODO` ?

**Review by TheSonOfThomp — APPROVED** [review]
LGTM! Just some small nits

**Pull request merged**